baseplate/server/__init__.py

```diff
@@ -197,3 +202,163 @@ def load_and_run_script():
     config = read_config(args.config_file, server_name=None, app_name=args.app_name)
     configure_logging(config, args.debug)
     args.entrypoint(config.app)
+
+
+def load_app_and_run_shell():
+    """Launch a shell."""
+    # modified from r2.commands:ShellCommand, but with support for thrift
+    # services
+    parser = argparse.ArgumentParser(
+        description="Open a shell with app configuration loaded.",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+
+    parser.add_argument("--debug", action="store_true", default=False,
+        help="enable extra-verbose debug logging")
+    parser.add_argument("--app-name", default="main", metavar="NAME",
+        help="name of app to load from config_file (default: main)")
+    parser.add_argument("config_file", type=argparse.FileType("r"),
+        help="path to a configuration file")
+
+    args = parser.parse_args(sys.argv[1:])
+    config = read_config(args.config_file, server_name=None, app_name=args.app_name)
+    configure_logging(config, args.debug)
+
+    locs = dict(__name__="pylons-admin")
+
+    is_http_service = pylons.config['pylons.package'] is not None
```

Comment:
Is there a better way to do this?

Reply:
The only reddit service using pylons is r2 and that has `reddit-shell` already. New HTTP services use Pyramid which has `pshell`. We really only need a custom shell for Thrift services.